sonar-html-plugin/src/main/java/org/sonar/plugins/html/checks/accessibility/FocusableInteractiveElementsCheck.java

```diff
@@ -0,0 +1,102 @@
+import static org.sonar.plugins.html.api.HtmlConstants.PRESENTATION_ROLES;
+import static org.sonar.plugins.html.checks.accessibility.AccessibilityUtils.isDisabledElement;
+import static org.sonar.plugins.html.checks.accessibility.AccessibilityUtils.isHiddenFromScreenReader;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.sonar.check.Rule;
+import org.sonar.plugins.html.checks.AbstractPageCheck;
+import org.sonar.plugins.html.node.TagNode;
+
+@Rule(key = "S6852")
+public class FocusableInteractiveElementsCheck extends AbstractPageCheck {
+
+  private static final String MESSAGE_TEMPLATE = "Elements with the \"%s\" interactive role must be focusable.";
+
+  private static final Set<String> INTERACTIVE_PROPS = new HashSet<>();
+  static {
+    INTERACTIVE_PROPS.addAll(EventHandlers.EVENT_HANDLERS_BY_TYPE.get("keyboard"));
+    INTERACTIVE_PROPS.addAll(EventHandlers.EVENT_HANDLERS_BY_TYPE.get("mouse"));
+  }
+
+  @Override
+  public void startElement(TagNode element) {
+    var role = element.getAttribute("role");
+    if (role == null) {
+      return;
+    }
+
+    if (!isHTMLTag(element)
+      || !hasInteractiveProps(element)
+      || !hasInteractiveRole(element)
+      || isDisabledElement(element)
+      || isHiddenFromScreenReader(element)
+      || isInteractiveElement(element)
+      || isNoninteractiveElement(element)
+      || hasPresentationRole(element)
+      || hasNoninteractiveRole(element)
+      || element.hasProperty("tabindex")
+      ) {
+      return;
+    }
+
+    var message = String.format(MESSAGE_TEMPLATE, role);
+    createViolation(element.getStartLinePosition(), message);
+  }
+
+  private static boolean isHTMLTag(TagNode element) {
+    return KNOWN_HTML_TAGS.stream().anyMatch(tag -> tag.equalsIgnoreCase(element.getNodeName()));
+  }
+
+  private static boolean hasInteractiveProps(TagNode element) {
+    return INTERACTIVE_PROPS.stream().anyMatch(prop -> {
+      var attr = element.getAttribute(prop);
+      return attr != null && !attr.isEmpty();
+    });
+  }
+
+  private static boolean hasInteractiveRole(TagNode element) {
+    return INTERACTIVE_ROLES.stream().anyMatch(role -> role.equalsIgnoreCase(element.getAttribute("role")));
+  }
+
+  private static boolean isInteractiveElement(TagNode element) {
+    return INTERACTIVE_ELEMENTS.stream().anyMatch(tag -> tag.equalsIgnoreCase(element.getNodeName()));
+  }
+
+  private static boolean isNoninteractiveElement(TagNode element) {
+    return NON_INTERACTIVE_ELEMENTS.stream().anyMatch(tag -> tag.equalsIgnoreCase(element.getNodeName()));
+  }
+
+  private static boolean hasPresentationRole(TagNode element) {
+    return PRESENTATION_ROLES.stream().anyMatch(role -> role.equalsIgnoreCase(element.getAttribute("role")));
+  }
+
+  private static boolean hasNoninteractiveRole(TagNode element) {
+    return NON_INTERACTIVE_ROLES.stream().anyMatch(role -> role.equalsIgnoreCase(element.getAttribute("role")));
+  }
```

Comment:
could you refactor this to use the methods from `HtmlConstants.java`, and move the new ones there?